api/envoy/config/rbac/v2alpha/rbac.proto

```diff
@@ -60,96 +67,76 @@ message RBAC {
-}
+  oneof rule {
+    option (validate.required) = true;
 
-// Specifies how to match ports.
-message PortMatch {
-  // Port numbers.
-  repeated uint32 ports = 1;
-}
+    // A set of rules that all must match in order to define the action.
+    Set and_rules = 1;
 
-// Permission defines a permission to access the service.
-message Permission {
-  // Optional. A list of HTTP paths or gRPC methods.
-  // gRPC methods must be presented as fully-qualified name in the form of
-  // packageName.serviceName/methodName.
-  // If this field is unset, it applies to any path.
-  repeated envoy.type.StringMatch paths = 1;
-
-  // Required. A list of HTTP methods (e.g., "GET", "POST").
-  // If this field is unset, it applies to any method.
-  repeated string methods = 2;
-
-  // Definition of a custom condition.
-  message Condition {
-    oneof condition_spec {
-      // Header match. This matches to the "request.http.headers" field in
-      // ":ref: `AttributeContext <envoy_api_msg_service.auth.v2alpha.AttributeContext>`.
-      // The map key is the header name. The header specifies how the service is accessed.
-      MapEntryMatch header = 1;
-
-      // Destination IP addresses.
-      IpMatch destination_ips = 2;
-
-      // Destination ports.
-      PortMatch destination_ports = 3;
-    }
-  }
+    // A set of rules where at least one must match in order to define the action.
+    Set or_rules = 2;
 
-  // Optional. Custom conditions.
-  repeated Condition conditions = 3;
+    // When any is set, it matches any action.
+    bool any = 3 [(validate.rules).bool.const = true];
+
+    // A header (or psuedo-header such as :path or :method) on the incoming HTTP request.
+    envoy.api.v2.route.HeaderMatcher header = 4;
+
+    // A CIDR block that describes the destination IP.
+    envoy.api.v2.core.CidrRange destination_ip = 5;
+
+    // A port number that describes the destination port connecting to.
+    uint32 destination_port = 6 [(validate.rules).uint32.lte = 65535];
+  }
 }
 
 // Principal defines an identity or a group of identities.
 message Principal {
+
+  message Set {
+    repeated Principal ids = 1 [(validate.rules).repeated.min_items = 1];
+  }
+
   // Authentication attributes for principal. These could be filled out inside RBAC filter.
   // Or if an authentication filter is used, they can be provided by the authentication filter.
   message Authenticated {
-    // Optional. The name of the principal. This matches to the "source.principal" field in
-    // ":ref: `AttributeContext <envoy_api_msg_service.auth.v2alpha.AttributeContext>`.
-    // If unset, it applies to any user.
+    // The name of the principal. If set, the URI SAN is used from the certificate, otherwise the
+    // subject field is used. If unset, it applies to any user that is authenticated.
     string name = 1;
   }
 
-  // Optional. Authenticated attributes that identify the principal.
-  Authenticated authenticated = 1;
+  oneof identifier {
```

Comment:
it was removed because it's source is unclear from Envoy's perspective. To get the same effect as what was previously described in the comment, use a header matcher on the "x-envoy-downstream-cluster"